aws-samples/aws-amplify-stepfunctions-example
# Migration to AWS Amplify Gen 2

*Issue #125 :*

*Description of changes:*
- Migrate 'amplify add api' to use Gen 2 backend in-build data
- Migrate 'amplify add custom' to use Gen 2 Custom CDK Construct
- Pass GraphQL API ID from Step 1 to be a dependency for Step 2 and simplify the code
- Change frontend code to call amplify-data client to trigger AppSync API using mutations.

To Test
1. Clone the Repo
2. Go to the repo folder
3. Run 'npm create amplify@latest'
4. Replace the files in amplify/ folder with the ones supplied under REPO/template/ folder.
5. In amplify/backend.ts file replace the place holder "YOUR_EMAIL_ID" with your email address.
6. Deploy the resources in Gen 2 sandbox by running 'npx ampx sandbox'
7. Accept the SNS Subscription for sentiment notifications from your email.
8. In Another window change directory to repo folder
9. Run 'npm run dev'
10. Open browser and go to http://localhost:8080 and test both negative and positive sentiments

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
